### ISSUE TITLE
Deprecate Stack in favor of Columns

### ISSUE THREAD

**Issue by ramnes** [qtile/qtile#1330]
The Columns layout implements everything Stack does, and more, so there's not point in keeping both. See #1251 for rationale (and for a potential future improvement of Columns).

**Comment by osimplex** [comment]
I don't agree. Just because the Stack Layout is simpler to configure.

How can I configure the Columns to behave as Stack with these defaults in `config.py`:
```python
layout_theme = {
	"margin":		2,
	"border_width":		2,
	"border_focus":		Colors.blue[0],
	"border_normal":	Colors.black[0],
}
```
Colums have configs that make it more difficult to get an consistent theme in terms of colors with an dict that I pass to all other layouts.
Also, the default behavior when there is only one column is not the same and, last but not least, the comands `layout.next()` and `layout.prev()` not behave as in Stack, they switch between windows instead of columns when split is `True`.

I use Stack+Max layouts for, example, an group where I put all pdf reader windows (is an good layout for reading). I use also for group of CLI programs as Irssi.
Indeed, I think could be good to have a pair as in Xmonad (MonadTall and MonadWide), StackTall and StackWide (good for some situations where I have terminals without tabs (and don't want to extent the terminal or use an external resource) and also for reading). Maybe I could try to write this, reading the `stack.py` and `xmonad.py`.

**Comment by osimplex** [comment]
I don't agree. Just because the Stack Layout is simpler to configure.

How can I configure the Columns to behave as Stack with these defaults in `config.py`:
```python
layout_theme = {
	"margin":		2,
	"border_width":		2,
	"border_focus":		Colors.blue[0],
	"border_normal":	Colors.black[0],
}
```
Colums have configs that make it more difficult to get an consistent theme in terms of colors with an dict that I pass to all other layouts.
Also, the default behavior when there is only one column is not the same and, last but not least, the comands `layout.next()` and `layout.prev()` not behave as in Stack, they switch between windows instead of columns when split is `True`.

I use Stack+Max layouts for, example, an group where I put all pdf reader windows (is an good layout for reading). I use also for group of CLI programs as Irssi.
Indeed, I think could be good to have a pair as in Xmonad (MonadTall and MonadWide), StackTall and StackWide (good for some situations where I have terminals without tabs (and don't want to extent the terminal or use an external resource) and also for reading). Maybe I could try to write this, reading the `stack.py` and `xmonad.py`.

I use Max, Floating, the Xmonad pair, Stack and Zoomy (for and group of "minimized" windows, but any layout fits well here). I think that I can replace MonadWide with a hipotetical StackWide layout.

**Comment by ramnes** [comment]
Alright, let me rephrase the title. :slightly_smiling_face: 

The real issue isn't Stack against Columns, but rather that we have a lot of different layouts with very similar functionalities but different codebases. The result is, as you describe, that they don't behave the same way when they could (read "should").

We can keep the different classes with different defaults and simpler or harder configurations, I don't mind about this; what I want to do is to reduce the number of layouts codebases, and thus here we could very well re-implement Stack based on Columns, while still keeping the simplicity of Stack and extensibility of Columns.

In the long term, reducing the number of layout codebases would avoid us issues such as #1251.

**Comment by osimplex** [comment]
"In the long term, reducing the number of layout codebases would avoid us issues such as `#1251`."
This is right.

**Comment by ramnes** [comment]
Doing so, we could remove `_ClientList`, which is only used in these two layouts.
